scientific-computing-with-python/learn-encapsulation-by-drawing-a-projectile-trajectory/projectile-trajectory.py

```diff
@@ -0,0 +1,153 @@
+        return f'angle: {round(math.degrees(self.__angle))}°\nspeed: {self.__speed} m/s\nstarting height: {self.__height} m\ndisplacement: {round(self.__calculate_displacement(), 1)} m'
+
+    def __str__(self):
+        return self.details()
+    
+    @property
+    def height(self):
+        return self.__height
+    
+    @height.setter
+    def height(self, new_height):
+        self.__height = new_height
+
+    @property
+    def angle(self):
+        return self.__angle
+    
+    @angle.setter
+    def angle(self, new_angle):
+        self.__angle = math.radians(new_angle)
+
+    @property
+    def speed(self):
+        return self.__speed
+
+    @speed.setter
+    def speed(self, new_speed):
+        self.__speed = new_speed
+
+class Graph:
+    def __init__(self, coordinates):
+        self.__coordinates = coordinates
+
+    def __create_coordinates_table(self):
+        table = '  x      y\n'
+        for x, y in self.__coordinates:
+            table += f'{x:>3}{round(y, 2):>7.2f}\n'
+
+        return table
+
+    def __create_trajectory(self):
+        rounded_coord = [(x, round(y)) for x, y in self.__coordinates]
+        rows = max([y for x, y in rounded_coord]) + 1
+        columns = max([x for x, y in rounded_coord]) + 1
+        graph_matrix = [[PROJECTILE if (x, y) in rounded_coord else ' ' for x in range(columns)] for y in range(rows, -1, -1)]
+        graph_rows = ['⊣' + ''.join(row) for row in graph_matrix] + [' ' + '⊤'*columns]
+        graph = '\n'.join(graph_rows)
+        return graph
+    
+    def __str__(self):
+        return self.__create_trajectory()
+
+    def table(self):
+        return self.__create_coordinates_table()
+
+def terminal_menu():
+    page = 0
+    while True:
+        if page == 0:
+            value_from_user = input(
+                "Please provide starting height, speed and angle for the PROJECTILE separated by a space: "
+            )
+            numbers = [int(number) for number in value_from_user.split()]
+            
+            bullet = Projectile(*numbers)
+            page = 1
+        elif page == 1:
+            print()
+            bullet.details()
+            try:
+                page = int(
+                    input(
+                        "\nChoose what to do:\n1. Draw trajectory\n2. List coordinates\n3. Change a value\n4. Exit\n"
+                    )) + 1
+            except:
+                print('Invalid input, please try again')
+        elif page == 2:
+            graph = Graph(bullet.calculate_all_coordinates())
+            print(graph)
+            page = 1
+        elif page == 3:
+            graph = Graph(bullet.calculate_all_coordinates())
+            print(graph.table())
+            page = 1
+        elif page == 4:
+            key = input(
+                'Write "speed", "angle", or "height" to choose which value: ')
+            try:
+                value = float(input('What value do you want to give? '))
+                if key == 'speed':
```

Comment:
can this be made less repetitive? is there something in python that allows that?